Add documentation for LG ThinQ integration

LG-ThinQ-Integration opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
Add documentation for LG ThinQ integration


## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [x] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/123860
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/brands/pull/5701
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **New Features**
	- Introduced LG ThinQ integration, enabling control of LG appliances, webOS TVs, and soundbars through Home Assistant.
	- Added comprehensive setup guide and functionality overview for various LG devices, including air conditioners, refrigerators, and washers.
	- Support for multiple device types and troubleshooting sections to enhance user experience.

- **Documentation**
	- Detailed instructions for obtaining a personal access token (PAT) and configuring devices.
	- Guidance on connecting webOS devices over a local network and device discovery methods.
	- Information on common issues, debugging tips, and requirements for successful integration.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

JangwonLeeLG committed:
Add documentation for lgthinq integration

frenck commented on source/_integrations/lgthinq.markdown:
```diff
@@ -0,0 +1,188 @@
+ha_config_flow: true
+ha_domain: lgthinq
+ha_platforms:
+  - binary_sensor
+  - button
+  - climate
+  - fan
+  - number
+  - select
+  - sensor
+  - switch
+  - text
+  - vacuum
+  - event
```
It is not possible to have all platforms in the initial PR.

Please limit PRs to a single platform per PR.

LG-ThinQ-Integration replied:
@frenck 
The initial code PR will have a single platform. Do you mean that I need to make a documentation PR for every code PR with a single platform?

frenck replied:
Yup. Every core PR affecting documentation needs its own documentation PR linked.

LG-ThinQ-Integration replied:
@frenck 
I limited this PR to a single platform.

frenck reviewed (dismissed):
This PR is not linked to a parent PR in our core repository.

../Frenck

LG-ThinQ-Integration committed:
Merge branch 'next' into next

LG-ThinQ-Integration committed:
Merge branch 'home-assistant:next' into next

LG-ThinQ-Integration commented:
@frenck 
Can I open this PR first, event though the code PR is a draft now?

LG-ThinQ-Integration committed:
Update source/_integrations/lgthinq.markdown

Accept the commit suggestion

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

LG-ThinQ-Integration commented on source/_integrations/lgthinq.markdown:
```diff
@@ -0,0 +1,178 @@
+- Washcombo Main
+- Washcombo Mini
+- [Washer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/washer/)
+- [Washtower](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/WashTower-Single-Unit/)
```
[Washtower] and [Washtower Dryer] are different devices. No need to modify it.

LG-ThinQ-Integration committed:
Update source/_integrations/lgthinq.markdown

Accept the suggestion.

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

LG-ThinQ-Integration committed:
Update source/_integrations/lgthinq.markdown

Accept the suggestion

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

LG-ThinQ-Integration commented on source/_integrations/lgthinq.markdown:
```diff
@@ -0,0 +1,178 @@
+
+Support LG Appliances as follows:
+
+- [Air Conditioner](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/air-conditioner/)
+- [Air Purifier](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/air-purifier/)
+- [Air Purifier Fan](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Air-Purifier-Fan/)
```
[Air Purifier] and [Air Purifier Fan] are different devices. No need to modify it.

JangwonLeeLG committed:
remove instructions about Soundbar and webOS TV

LG-ThinQ-Integration committed:
Merge branch 'next' into next

LG-ThinQ-Integration committed:
Merge branch 'next' into next

LG-ThinQ-Integration committed:
Merge branch 'next' into next

LG-ThinQ-Integration committed:
Update lgthinq.markdown

LG-ThinQ-Integration committed:
Merge branch 'next' into next

LG-ThinQ-Integration commented:
> This PR is not linked to a parent PR in our core repository.
> 
> ../Frenck

I linked this PR to a parent PR.
An the parent PR has been merged.

joostlek commented on source/_integrations/lgthinq.markdown:
```diff
@@ -0,0 +1,136 @@
+featured: true
+ha_category:
+  - Hub
+ha_release: 2024.7.3
```
```suggestion
ha_release: "2024.10"
```

LG-ThinQ-Integration replied:
Edited

joostlek commented on source/_integrations/lgthinq.markdown:
```diff
@@ -0,0 +1,136 @@
+ha_code_owners:
+  - '@LG-ThinQ-Integration'
+ha_config_flow: true
+ha_domain: lgthinq
```
```suggestion
ha_domain: lg_thinq
```

joostlek reviewed (dismissed):
Please rename the file to `lg_thinq`

LG-ThinQ-Integration committed:
Update source/_integrations/lgthinq.markdown

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

LG-ThinQ-Integration committed:
Update source/_integrations/lgthinq.markdown

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

LG-ThinQ-Integration committed:
Rename lgthinq.markdown to lg_thinq.markdown

LG-ThinQ-Integration commented:
> Please rename the file to `lg_thinq`

I renamed the file and modified what you suggested.
As I see, there is one more thing to change, am I right?

```yaml
logger:
  default: info
  logs:
    homeassistant.components.lgthinq: debug
```

LG-ThinQ-Integration committed:
Update source/_integrations/lg_thinq.markdown

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

LG-ThinQ-Integration committed:
Merge branch 'next' into next

LG-ThinQ-Integration committed:
Update lg_thinq.markdown

LG-ThinQ-Integration commented:
@joostlek 
In the manifest of the code (lg_thinq), document path is 'https://www.home-assistant.io/integrations/lgthinq/' 
Do I need to change it with a new PR? Or can I change it in the next entity PR?

joostlek commented:
Oh yea we should change that as well

LG-ThinQ-Integration commented:
> Oh yea we should change that as well

Can I change that in this bugfix PR together? Or a new PR?
- https://github.com/home-assistant/core/pull/125052

joostlek commented:
Let's add that to the binary sensor PR

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,136 @@
+ha_integration_type: integration
+---
+
+The LG ThinQ integration allows you to connect LG ThinQ devices to Home Assistant. The features of this integration include:
```
```suggestion
The **LG ThinQ** integration allows you to connect LG ThinQ devices to Home Assistant. The features of this integration include:
```

joostlek commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,136 @@
+4. Click the **CREATE TOKEN** button.
+5. Once all the steps are completed, you will see that a **PAT** has been generated.
+
+### MQTT
+
+The LG ThinQ integration connects to the LG ThinQ Cloud via MQTT and receives push status updates from the devices. This is a part of the LG ThinQ API implementation and is automatically connected through internal operations, so no additional settings are required.
+
+{% note %} MQTT connections are only supported in the Appliance feature.{% endnote %}
```
This is not in the current integration

LG-ThinQ-Integration replied:
Removed

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,136 @@
+The LG ThinQ integration allows you to connect LG ThinQ devices to Home Assistant. The features of this integration include:
+
+- Control LG appliances as Home Assistant entities through the [LG ThinQ Connect API](https://thinq.developer.lge.com/ko/cloud/).
+- This feature works via LG ThinQ cloud; therefore, an internet connection is required.
```
```suggestion
```
I think this is better placed in the Prereq section

joostlek commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,136 @@
+2. Choose the region (country).
+
+
+## Delete
+
+To remove the integration, please follow these steps:
+
+1. Go to the Home Assistant Integrations page.
+2. Select the LG ThinQ integration instance.
+3. Click on the options menu and choose Delete.
+
+{% note %} Deleting the integration will remove all entities associated with your LG ThinQ devices' features. {% endnote %}
```
Not sure if this is LG specific

c0ffeeca7 replied:
Good point. we should add a general procedure that integration docs can point to, and then add the integration specific information only.

LG-ThinQ-Integration replied:
Do you mean we don't need it here because it is not LG specific? 
Then let me remove it.

LG-ThinQ-Integration replied:
Removed.

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,136 @@
+
+## Prerequisites
+
+- An internet connection and a [personal access token](https://connect-pat.lgthinq.com) are required to use the [LG ThinQ Connect API](https://thinq.developer.lge.com/ko/cloud/).
```
```suggestion
- This feature works via LG ThinQ cloud, therefore an internet connection is required.
- A [personal access token](https://connect-pat.lgthinq.com) to use the [LG ThinQ Connect API](https://thinq.developer.lge.com/ko/cloud/).
```

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,136 @@
+### Personal Access Token (PAT)
+
+1. Access the **[personal access token](https://connect-pat.lgthinq.com)** page (requires an LG ThinQ account).
+2. Click the **ADD NEW TOKEN** button.
```
```suggestion
2. Select **ADD NEW TOKEN**.
```

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,136 @@
+1. Access the **[personal access token](https://connect-pat.lgthinq.com)** page (requires an LG ThinQ account).
+2. Click the **ADD NEW TOKEN** button.
+3. Enter a new token name and select the following authorized scopes: 
+    - Permission to view all devices
+    - Permission to view all device statuses
+    - All device control rights
+    - All device event subscription rights
+    - All device push notification permissions
```
```suggestion
    - **Permission to view all devices**
    - **Permission to view all device statuses**
    - **All device control rights**
    - **All device event subscription rights**
    - **All device push notification permissions**
```

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,136 @@
+    - All device control rights
+    - All device event subscription rights
+    - All device push notification permissions
+4. Click the **CREATE TOKEN** button.
```
```suggestion
4. Select **CREATE TOKEN**.
```

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,136 @@
+
+The LG ThinQ integration connects to the LG ThinQ Cloud via MQTT and receives push status updates from the devices. This is a part of the LG ThinQ API implementation and is automatically connected through internal operations, so no additional settings are required.
+
+{% note %} MQTT connections are only supported in the Appliance feature.{% endnote %}
```
```suggestion
{% note %} MQTT connections are only supported in the **Appliance** feature.{% endnote %}
```

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,136 @@
+{% note %} MQTT connections are only supported in the Appliance feature.{% endnote %}
+
+
+## Setup
+
+To set up the integration, please follow these steps:
+
```
```suggestion
```

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,136 @@
+2. Choose the region (country).
+
+
+## Delete
```
```suggestion
## Removing the LG ThinQ integration
```

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,136 @@
+
+## Delete
+
+To remove the integration, please follow these steps:
```
```suggestion
To remove the LG ThinQ integration, follow these steps:
```

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,136 @@
+
+To remove the integration, please follow these steps:
+
+1. Go to the Home Assistant Integrations page.
```
```suggestion
1. Go to **{% my integrations icon title="Settings > Devices & Services" %}**.
```
There's a my link for this page.

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,136 @@
+To remove the integration, please follow these steps:
+
+1. Go to the Home Assistant Integrations page.
+2. Select the LG ThinQ integration instance.
```
```suggestion
2. Select the LG ThinQ integration.
```

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,136 @@
+
+1. Go to the Home Assistant Integrations page.
+2. Select the LG ThinQ integration instance.
+3. Click on the options menu and choose Delete.
```
```suggestion
3. Select the three dots {% icon "mdi:dots-vertical" %} menu, then select **Delete**.
```

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,136 @@
+
+#### Aborted: The country is not supported
+
+Check your PAT's valid country in the "Choose the region / Country" section.
```
```suggestion
Check your PAT's valid country in the **Choose the region / Country** section.
```

c0ffeeca7 committed:
Tiny style tweaks

LG-ThinQ-Integration commented:
> Let's add that to the binary sensor PR

OK, I added.

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,132 @@
+#### Error: The number of API calls has been exceeded
+
+This error occurs when there's an abnormal number of API calls made using the PAT.
+
```
an recommendation what to do when that happens?

LG-ThinQ-Integration replied:
Actually no recommendation but waiting.
Then do we have to remove it?

c0ffeeca7 replied:
nono, that's fine. Just tell them to wait then. I think it's better if it's documented, even if it seems obvious.

LG-ThinQ-Integration replied:
Added it as below.

The LG ThinQ integration will work properly after some time.

LG-ThinQ-Integration committed:
Update lg_thinq.markdown

Remove Mqtt descriptions

LG-ThinQ-Integration committed:
Update source/_integrations/lg_thinq.markdown

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
````diff
@@ -0,0 +1,126 @@
+  default: info
+  logs:
+    homeassistant.components.lg_thinq: debug
+```
````
maybe we should add a step to remove the debug level once they are done debugging?

LG-ThinQ-Integration replied:
Do you mean we need to mention how to remove the debug level?
It is just removing the relevant line from the configuration.yaml.

LG-ThinQ-Integration replied:
@c0ffeeca7 
How can I resolve this one?
I can't find any other integrations to refer.
We just referred other integrations like smartthings.

LG-ThinQ-Integration committed:
Update lg_thinq.markdown

LG-ThinQ-Integration committed:
Update lg_thinq.markdown

LG-ThinQ-Integration commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,126 @@
+
+## Prerequisites
+
+- This feature works via LG ThinQ cloud. Therefore, an internet connection is required.
```
Applied.

LG-ThinQ-Integration committed:
Update lg_thinq.markdown

LG-ThinQ-Integration committed:
Update lg_thinq.markdown

LG-ThinQ-Integration commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,126 @@
+
+This error occurs when there's an abnormal number of API calls made using the PAT.
+
+
```
Removed

LG-ThinQ-Integration commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,126 @@
+- [Dryer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Dryer/)
+- [Home Brew](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Home-Brew/)
+- [Hood](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Hood/)
+- Humidifier
```
A note is added.

LG-ThinQ-Integration commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,126 @@
+- [Styler](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/styler/)
+- [System Boiler](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/System-Boiler/)
+- Washcombo Main
+- Washcombo Mini
```
A note is added.

LG-ThinQ-Integration committed:
Update lg_thinq.markdown

LG-ThinQ-Integration commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,116 @@
+- [Water Purifier](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/water-purifier/)
+- [Wine Cellar](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Wine-Cellar/)
+
+
```
Updated

LG-ThinQ-Integration committed:
Update lg_thinq.markdown

LG-ThinQ-Integration commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,110 @@
+- [Wine Cellar](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/wine-cellar/)
+{% note %}
+Humidifier, Washcombo Main and Washcombo Mini are also supported, but the links are not available yet.
+{% endnote %}
```
Updated

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
````diff
@@ -0,0 +1,106 @@
+  default: info
+  logs:
+    homeassistant.components.lg_thinq: debug
+```
````
```suggestion
```
Don't forget to remove the line again once you are done with debugging.
```

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,106 @@
+## Debugging
+
+The LG ThinQ integration will show additional information by enabling log configuration. Add the line to your {% term "`configuration.yaml`" %}:
+
```
```suggestion
Don't forget to remove the line again once you are done debugging.
```

c0ffeeca7 committed:
Update source/_integrations/lg_thinq.markdown

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -0,0 +1,106 @@
+## Debugging
+
+The LG ThinQ integration will show additional information by enabling log configuration. Add the line to your {% term "`configuration.yaml`" %}:
+Don't forget to remove the line again once you are done debugging.
```
```suggestion
Don't forget to remove the line again once you are done debugging.

```

c0ffeeca7 committed:
Update source/_integrations/lg_thinq.markdown

c0ffeeca7 reviewed (approved):
Thank you, @LG-ThinQ-Integration :+1:

LG-ThinQ-Integration commented:
@joostlek 
I guess I need your approval too?
